### ISSUE TITLE
Automate posting on social medias

### ISSUE THREAD

**Issue by dannycolin** [OpenScienceMOOC/site#27]
List of social medias when want to target: 

- facebook
- twitter
- mastodon
- ???

What we need: 

- Possibility to format post for each platform
- Limit post to specific platforms
- ???

Solution: 

Use jekyll to generate an RSS feed that we could use as an input source in ~~IFTTT.com or zapier.com~~ Huginn.

**Comment by Protohedgehog** [comment]
This sounds complex, do you need support for it at all..? :)

**Comment by dannycolin** [comment]
It's very similar to the blog section so It shouldn't be to hard. But, of course, help is always welcome :). For now, I'm still working on the website redesign so if you want to try here's some resources:

- https://jekyllrb.com/tutorials/convert-site-to-jekyll/#10-rss-feed
- https://www.w3schools.com/xml/xml_rss.asp

**Comment by tosteiner** [comment]
1. Open Community Calendar established on cal.opensciencemooc.eu
2. New event entries are currently triggering new posts on the Slack `#event` channel, this is facilitated via the external Zapier service.
3. With our plans to move from Slack to Mattermost, it'd be great if we could make use of that option via open source solution.

### Suggestion: 

* Set up self-hosted [Huginn](https://dev.to/heroku/huginn-an-open-source-self-hosted-ifttt-5hd6) instance next to Mattermost and Moodle.
* Link new events on cal.opensciencemooc.eu to `#events` channel in Mattermost
* Post new events on OSMOOC Twitter channel

**Comment by jcolomb** [comment]
Can't we just use zapier to send the same content to mattermost and twitter? probably not free then?

**Comment by tosteiner** [comment]
@jcolomb yup, sure - that would be the short-term solution :) so as to stay in control of our data in the longer term, Huginn would allow us to do this ourselves and cut back dependencies from third-party services 😉 but nothing that's mission-critical at the moment ;)
